app/src/main/java/ca/llamabagel/transpo/data/SearchRepository.kt

```diff
@@ -25,47 +31,46 @@ object OttawaBoundaries {
     const val MAX_LNG = -75.351405
 }
 
-class SearchRepository @Inject constructor(private val database: TransitDatabase, private val strings: StringsGen) {
+@Singleton
+@ExperimentalCoroutinesApi
+class SearchRepository @Inject constructor(
+    private val database: TransitDatabase,
+    private val strings: StringsGen,
+    private val dispatcher: CoroutinesDispatcherProvider
+) {
 
-    suspend fun getSearchResults(query: String): List<SearchResult> = withContext(Dispatchers.IO) {
-        if (query.isEmpty()) return@withContext emptyList<SearchResult>()
+    val routeChannel = ConflatedBroadcastChannel<List<SearchResult.RouteItem>>()
+    val stopChannel = ConflatedBroadcastChannel<List<SearchResult.StopItem>>()
+    val placeChannel = ConflatedBroadcastChannel<List<SearchResult.PlaceItem>>()
 
-        val searchResults = mutableListOf<SearchResult>()
-
-        getStops(query).takeIf { it.isNotEmpty() }?.let { stops ->
-            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_routes)))
-            searchResults.addAll(stops)
-        }
-
-        getRoutes(query).takeIf { it.isNotEmpty() }?.let { routes ->
-            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_routes)))
-            searchResults.addAll(routes)
-        }
-
-        getPlaces(query).takeIf { it.isNotEmpty() }?.let { places ->
-            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_places)))
-            searchResults.addAll(places)
+    suspend fun getSearchResults(query: String) {
+        if (query.isNotEmpty()) {
```

Comment:
All channels should emit empty lists if the query is empty.

Reply:
Eventually, a "recent" or "suggested" search list can be emitted if the query is empty.

Reply:
Some of the database throw errors if the query is empty

Reply:
The database queries don't need to be run if the string is empty, just emit empty lists.